channeldb/invoices.go

```diff
@@ -1290,12 +1318,20 @@ func (d *DB) updateInvoice(hash lntypes.Hash, invoices, settleIndex *bbolt.Bucke
 		if _, exists := invoice.Htlcs[key]; exists {
 			return nil, fmt.Errorf("duplicate add of htlc %v", key)
 		}
+
+		// Force caller to supply htlc without custom records in a
+		// consistent way.
+		if htlcUpdate.CustomRecords == nil {
```

Comment:
i guess this conflicts with what i said earlier about constructor returning nil, i'm fine if you prefer it to always be initialized.

Reply:
Yes, I wanted to prevent to case where a future dev would do a `nil` check and forget about the case where the map is empty.